routes/boxHistory.js

```diff
@@ -0,0 +1,147 @@
+const express = require('express');
+
+const boxRouter = express();
+const pgp = require('pg-promise')({});
+const { db } = require('../config');
+
+const cn = `postgresql://${process.env.REACT_APP_DATABASE_USER}:${process.env.REACT_APP_DATABASE_PASSWORD}@${process.env.REACT_APP_DATABASE_HOST}:${process.env.REACT_APP_DATABASE_PORT}/${process.env.REACT_APP_DATABASE_NAME}?ssl=true`; // For pgp
+
+const database = pgp(cn);
+
+boxRouter.use(express.json());
+
+const SQLQueries = {
+  UpdateBox: (
+    status,
+    boxHolderName,
+    boxHolderEmail,
+    zipCode,
+    generalLocation,
+    message,
+    changesRequested,
+    rejectionReason,
+    messageStatus,
+    dropOffMethod,
+  ) =>
+    `UPDATE "Box_History" SET
+        box_id = $(boxID)
+        ${status !== undefined ? ', status = $(status)' : ''}
+        ${boxHolderName !== undefined ? ', boxholder_name = $(boxHolderName)' : ''}
+        ${boxHolderEmail !== undefined ? ', boxholder_email = $(boxHolderEmail)' : ''}
+        ${zipCode !== undefined ? ', zip_code = $(zipCode)' : ''}
+        ${generalLocation !== undefined ? ', general_location = $(generalLocation)' : ''}
+        ${message !== undefined ? ', message = $(message)' : ''}
+        ${changesRequested !== undefined ? ', changes_requested = $(changesRequested)' : ''}
+        ${rejectionReason !== undefined ? ', rejection_reason = $(rejectionReason)' : ''}
+        ${messageStatus !== undefined ? ', message_status = $(messageStatus)' : ''}
+        ${dropOffMethod !== undefined ? ', drop_off_method = $(dropOffMethod)' : ''}
+        WHERE
+          box_id = $(boxID)`,
+  Return: 'Returning *',
+  GetBoxes: (pickup) =>
+    `SELECT * FROM "Box_History" WHERE
+      ($(status) = '' OR status = $(status))
+      ${pickup ? 'AND pickup = $(pickup)' : ''}`,
+  GetBox: 'SELECT * FROM "Box_History" WHERE box_id = $1',
+  CopyBoxInfoToAnchorBox:
+    'UPDATE "Anchor_Box" SET message = $1, zip_code = $2, picture = $3, general_location = $4, date=$5 WHERE box_id = $6',
+  ApproveBox: 'UPDATE "Box_History" SET approved = TRUE, status = \'evaluated\' WHERE box_id = $1',
+};
+
+// update status of pick up box
+boxRouter.put('/update', async (req, res) => {
+  try {
+    const {
+      status,
+      boxID,
+      boxHolderName,
+      boxHolderEmail,
+      zipCode,
+      generalLocation,
+      message,
+      changesRequested,
+      rejectionReason,
+      messageStatus,
+      dropOffMethod,
```

Comment:
Did we decide that we are changing this field to launchedOrganically instead? Let's finalize on either dropOffMethod or launchedOrganically and keep that consistent across all tables. I would prefer launchedOrganically to match the designs.